SimplenoteUITests/NoteEditor.swift

```diff
@@ -177,6 +185,12 @@ class NoteEditor {
         return matches
     }
 
+    class func openHistory() {
+        app.navigationBars[UID.NavBar.allNotes].buttons[UID.Button.noteEditorMenu].tap()
+        guard app.tables.staticTexts[UID.Text.nodeEditorOptionsHistory].exists else {return}
+        app.tables.staticTexts[UID.Text.nodeEditorOptionsHistory ].tap()
```

Comment:
Just a tricky question: why there's a guard statement for `History` button, but not for the `Menu` button? 😉 

Also, if we use the control more than once in a method, I see sense in having it as a constant:

```suggestion
        let historyButton = app.tables.staticTexts[UID.Text.nodeEditorOptionsHistory]
        guard historyButton.exists else { return }
        historyButton.tap()
```

Reply:
Ok, let's see if I got it right. =P
The `Menu` button is in the navigation bar, so we 'know' it is there while the `History` button is not visible yet and there's some animation in place. If `History` button does not exist the guard statement will stop the `openHistory()` flow without trying to tap the button and the test will gracefully fail the `historyShown()` assertion. If we tried to tap a button that does not exist it would just break the test execution.

Looking at it now I that using `historyButton.waitForExistence(minLoadTimeout)` would be more interesting if dealing with some slow devices/simulators. Does it make sense?

Reply:
I think there is no _right_ answer, since wrapping everything in guards is a questionable time investment, and you can see that it's quite sporadic in the existing code. I was just curious about your opinion 🙂 

> Looking at it now I that using historyButton.waitForExistence(minLoadTimeout) would be more interesting if dealing with some slow devices/simulators. Does it make sense?

We can never be extra careful, so why not 🙂 In practice, the animation of note Menu is fast enough to not cause any issues even on notoriously slow CI runners (we know this because there's a very similar `toggleMarkdownState()` method that does not use waits and did not result in flakiness).